mimic/model/flavor_objects.py

```diff
@@ -1,279 +1,107 @@
-                                    "disk": 40},
-               "4 GB Performance": {"id": "performance1-4", "ram": 4096, "vcpus": 4, "rxtx_factor": 800,
-                                    "disk": 40},
-               "8 GB Performance": {"id": "performance1-8", "ram": 8192, "vcpus": 8, "rxtx_factor": 1600,
-                                    "disk": 40}}
-
-    def extra_specs_json(self):
+    def get_flavor(self, http_get_request, flavor_id, absolutize_url):
         """
-        Create a JSON-serializable data structure describing
-        ``OS-FLV-WITH-EXT-SPECS:extra_specs`` for a performance1 flavor.
+        Return a flavor object if one exists from the list `/flavors` api,
+        else creates and adds the flavor to the :obj: `flavors_store`.
+        If the `flavor_id` is listed in `mimic.canned_responses.mimic_presets`,
+        then will return 404.
         """
-        return {
-            "class": "performance1",
-            "policy_class": "performance_flavor"
-        }
+        flavor = self.flavor_by_id(flavor_id)
+        if flavor is None:
+            return dumps(not_found("The resource could not be found.",
+                                   http_get_request))
+        return dumps({"flavor": flavor.detailed_json(absolutize_url)})
 
 
-class RackspacePerformance2Flavor(Flavor):
+@attributes(["tenant_id", "clock",
+             Attribute("regional_collections", default_factory=dict)])
+class GlobalFlavorCollections(object):
     """
-    A Rackspace performance flavor object representation
+    A :obj:`GlobalFlavorCollections` is a set of all the
+    :obj:`RegionalFlavorCollection` objects owned by a given tenant.  In other
+    words, all the flavor objects that a single tenant owns globally.
     """
-    flavors = {"15 GB Performance": {"id": "performance2-15", "ram": 15360, "vcpus": 4,
-                                     "rxtx_factor": 1250, "disk": 40},
-               "30 GB Performance": {"id": "performance2-30", "ram": 30720, "vcpus": 8,
-                                     "rxtx_factor": 2500, "disk": 40},
-               "60 GB Performance": {"id": "performance2-60", "ram": 61440, "vcpus": 16,
-                                     "rxtx_factor": 5000, "disk": 40},
-               "90 GB Performance": {"id": "performance2-90", "ram": 92160, "vcpus": 24,
-                                     "rxtx_factor": 7500, "disk": 40},
-               "120 GB Performance": {"id": "performance2-120", "ram": 122880, "vcpus": 32,
-                                      "rxtx_factor": 10000, "disk": 40}}
 
-    def extra_specs_json(self):
+    def collection_for_region(self, region_name):
         """
-        Create a JSON-serializable data structure describing
-        ``OS-FLV-WITH-EXT-SPECS:extra_specs`` for a performance 2 flavor.
+        Get a :obj:`RegionalFlavorCollection` for the region identified by the
+        given name.
```

Comment:
OK, looks like this comes in from the `NovaRegion` object and it would be difficult or impossible to set the name of this to whatever you want, so this code is chill.